test/e2e/test_helm_install_upgrade.go

```diff
@@ -466,3 +459,42 @@ func gatewayDataPlaneDeploymentIsNotPatched(gatewayLabelSelector string) func(co
 		}
 	}
 }
+
+func clusterWideResourcesAreProperlyManaged(gatewayLabelSelector string) func(ctx context.Context, c *assert.CollectT, cl client.Client) {
+	return func(ctx context.Context, c *assert.CollectT, cl client.Client) {
+		gw := getGatewayByLabelSelector(gatewayLabelSelector, ctx, c, cl)
+		controlplanes, err := gateway.ListControlPlanesForGateway(ctx, cl, gw)
+		if err != nil {
+			c.Errorf("failed to list ControlPlanes for Gateway %q: %v", client.ObjectKeyFromObject(gw), err)
+			c.FailNow()
+		}
+		require.Len(c, controlplanes, 1)
```

Comment:
Given that those assertions are run inside the `require.EventuallyWithT` I believe we want to use `assert` + return instead of `require` so that we don't abort the whole test right away but we continue to try to satisfy the assertion.